src/hotspot/share/classfile/symbolTable.cpp

```diff
@@ -91,7 +91,12 @@ static volatile bool   _has_items_to_clean = false;
 
 
 static volatile bool _alt_hash = false;
+
+#ifdef USE_LIBRARY_BASED_TLS_ONLY
 static volatile bool _lookup_shared_first = false;
+#else
+static THREAD_LOCAL bool _lookup_shared_first = false;
```

Comment:
Can you add an explanatory comment as to why this is a thread-local please. Just a couple of lines. Thanks.